cylc/flow/task_proxy.py

```diff
@@ -474,8 +495,40 @@ def state_reset(
     ) -> bool:
         """Set new state and log the change. Return whether it changed."""
         before = str(self)
+        if status == TASK_STATUS_EXPIRED:
+            is_queued = False
         if self.state.reset(status, is_held, is_queued, is_runahead):
-            if not silent:
+            if not silent and not self.transient:
                 LOG.info(f"[{before}] => {self.state}")
             return True
         return False
+
+    def satisfy_me(self, prereqs) -> bool:
+        """Attempt to satisfy the given prerequisites.
+
+        return True if all are valid, else False.
+        """
+        bad = self.state.satisfy_me(prereqs)
+        for err in bad:
+            LOG.warning(
+                f"{self.identity} does not depend on"
+                f" {err[0]}/{err[1]}:{err[2]}"
+            )
+        return len(bad) == 0
```

Comment:
Can save parsing/formatting IDs by passing through the Tokens object e.g:

```suggestion
    def satisfy_me(self, tokens: Tokens) -> bool:
        """Attempt to satisfy the given prerequisites.

        return True if all are valid, else False.
        """
        bad = self.state.satisfy_me(tokens)  # TODO
        for err in bad:
            LOG.warning(
                f"{self.identity} does not depend on"
                f" {tokens.relative_id_with_selectors}"
            )
        return len(bad) == 0
```